bundles/org.openhab.binding.miele/src/main/java/org/openhab/binding/miele/internal/TimeStabilizer.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.miele.internal;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Deque;
+import java.util.Iterator;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * {@link TimeStabilizer} keeps a cache of historic timestamp values in order to
+ * provide moving average calculations to smooth out short-term fluctuations.
+ *
+ * @author Jacob Laursen - Initial contribution
+ */
+@NonNullByDefault
+public class TimeStabilizer {
+
+    private final static int SLIDING_SECONDS = 300;
+    private final static int MAX_FLUCTUATION_SECONDS = 180;
+
+    private final Deque<Item> cache = new ConcurrentLinkedDeque<Item>();
+
+    private class Item {
+        public Instant start;
+        public Instant end;
+        public Instant instant;
+
+        public Item(Instant instant, Instant start, Instant end) {
+            this.start = start;
+            this.end = end;
+            this.instant = instant;
+        }
+    }
+
+    public TimeStabilizer() {
+    }
+
+    public void clear() {
+        cache.clear();
+    }
+
+    public Instant apply(Instant value) {
+        return this.apply(value, Instant.now());
+    }
+
+    public Instant apply(Instant value, Instant now) {
+        if (cache.isEmpty()) {
+            cache.add(new Item(value, now, now));
+            return value;
+        }
+
+        @Nullable
+        Item first = cache.getFirst();
+        @Nullable
+        Item last = cache.getLast();
+        last.end = now;
```

Comment:
Maybe you should check that `first` and `last` are not null before using them ?

Reply:
It is checked a few lines above by isEmpty.

Reply:
That's right.